backend/de.metas.business/src/main/java/de/metas/shipping/process/C_Order_AddTo_M_ShipperTransportation.java

```diff
@@ -53,15 +59,24 @@ public ProcessPreconditionsResolution checkPreconditionsApplicable(@NonNull fina
 	{
 		if (context.isNoSelection())
 		{
-			return ProcessPreconditionsResolution.rejectBecauseNoSelection();
+			return ProcessPreconditionsResolution.rejectBecauseNotSingleSelection();
 		}
 
-		final OrderId orderId = OrderId.ofRepoId(context.getSingleSelectedRecordId());
-		final org.compiere.model.I_C_Order order = orderBL.getById(orderId);
-		if (orderBL.isRequisition(order))
+		final IQueryFilter<I_C_Order> queryFilter = context.getQueryFilter(I_C_Order.class);
+		final List<I_C_Order> selectedOrders = orderBL.getByQueryFilter(queryFilter);
+
+		if (selectedOrders.stream().anyMatch(orderBL::isRequisition))
 		{
 			return ProcessPreconditionsResolution.rejectWithInternalReason("is purchase requisition");
 		}
+		if (selectedOrders.stream().anyMatch(o -> !orderBL.isCompleted(o)))
+		{
+			return ProcessPreconditionsResolution.reject(MSG_DocumentNotComplete);
+		}
+		if (shipperTransportationBL.isOrderLineAssignedToDifferentTransportationOrder(p_M_ShipperTransportation_ID, queryFilter))
```

Comment:
pls void having DAO/repo methods with IQueryFilter parameter (ofc, if u can).
In this case you can avoid it because you've already loaded the orders => so passing a Set<OrderId> would be feasible.